### ISSUE TITLE
Flathub strips out captions from appstream data

### ISSUE THREAD

**Issue by bbhtt** [flathub-infra/flatpak-builder-lint#280]
For some reason Flathub, server side is stripping out captions from appstream catalogue data. The same is not reproducible when building locally using the exact same commands and `org.flatpak.Builder`

The caption check should be re-enabled once the catalogue data has captions, otherwise the check has no meaning

See also https://github.com/flathub-infra/website/issues/48

**Comment by bbhtt** [comment]
I'm disabling the check for now.

**Comment by hadess** [comment]
Ran into this building https://github.com/flathub/com.github.shonumi.gbe-plus/, which would complain about `appstream-screenshot-missing-caption` even though I just [added captions](https://github.com/flathub/com.github.shonumi.gbe-plus/blob/master/com.github.shonumi.gbe-plus.appdata.xml#L22).

Locally, it doesn't complain about it:
```sh
$ flatpak run --command=flatpak-builder-lint org.flatpak.Builder builddir _build/
{
    "errors": [
        "appid-uses-code-hosting-domain"
    ],
    "warnings": [
        "finish-args-redundant-device-all"
    ]
}

Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter
```

**Comment by bbhtt** [comment]
Yea that's what confused me too. Clearly, something is going wrong on the server, but I don't have the power to debug that.

**Comment by bbhtt** [comment]
Solved by porting to libappstream, will re-enable the check once more apps get rebuilt and composes their metadata with libappstream

```
cat appstream.xml|grep "<caption>"
        <caption>Screnshoot 1</caption>
        <caption>Classic View</caption>
        <caption>RHDC View</caption>
        <caption>Image of Arduino IDE v2</caption>
        <caption>Simple view</caption>
        <caption>Advanced view</caption>
        <caption>Recording page</caption>
        <caption>Talk on an invite only server</caption>
        <caption>Voice and Video</caption>
        <caption>Dark Mode Window</caption>
        <caption>Light Mode Window</caption>
        <caption>Available on Linux, macOS, Windows, iOS, Android, and your web browser</caption>
        <caption>Running the benchmark</caption>
        <caption>CLEAR AND SIMPLE UI</caption>
        <caption>PERFECT FOR ORIGINAL PRUSA</caption>
        <caption>BUILT-IN 3RD PARTY PRINTER PROFILES</caption>
        <caption>MSLA AND MULTI MATERIAL SUPPORT</caption>
        <caption>AUTO-UPDATING PROFILES</caption>
        <caption>FIRMWARE FLASHER</caption>
        <caption>G-CODE VIEWER</caption>
        <caption>Application showcase</caption>
        <caption>Print preparation screen</caption>
        <caption>Editing TypeScript and searching for extensions</caption>
        <caption>YACReader &amp; YACReaderLibrary</caption>
        <caption>Properties</caption>
        <caption>The Library page</caption>
        <caption>The Library page with the category sidebar open</caption>
        <caption>The Preferences page (Library settings)</caption>
        <caption>The Explorer page in mobile format (Servers list)</caption>
        <caption>The Details page</caption>
        <caption>The Details page (chapters list)</caption>
        <caption>The Reader page</caption>
        <caption>The Reader page in mobile format</caption>
        <caption>The Category editor page in mobile format</caption>
        <caption>The Reading history page in mobile format</caption>
        <caption>Light Theme</caption>
        <caption>Dark Theme</caption>
        <caption>A simple tag search</caption>
        <caption>Media viewer</caption>
        <caption>The thread watcher works for 4chan and 8chan</caption>
        <caption>Gallery downloader</caption>
        <caption>Display and manage thousands of files</caption>
        <caption>Files can be sorted and collected by their tags</caption>
        <caption>Media viewer in fullscreen</caption>
        <caption>Many file formats are supported</caption>
        <caption>You can run your own (simple!) booru</caption>
        <caption>YesPlayMusic&apos;s Library Page</caption>
        <caption>Stimulator light screenshot</caption>
        <caption>Stimulator dark screenshot</caption>
        <caption>Stimulator light screenshot</caption>
        <caption>Stimulator dark screenshot</caption>
        <caption>Main window</caption>
        <caption>Main window with tabs</caption>
        <caption>Main Window</caption>
        <caption>Main Window Dark</caption>
        <caption>Preferences</caption>
        <caption>Adaptive Main Window</caption>
        <caption>RPCS3 running a game using the Vulkan backend</caption>
        <caption>JMeter screen</caption>
        <caption>Results aggregate graph</caption>
        <caption>Results response time graph</caption>
        <caption>The Gaphor main window</caption>
        <caption>Sequence diagram</caption>
        <caption>Customizable styling</caption>
        <caption>Dark mode</caption>
        <caption>Placing a call</caption>
        <caption>The call history</caption>
        <caption>Homepage</caption>
        <caption>Menu Actions</caption>
        <caption>Tabs View</caption>
        <caption>Desktop User Interface</caption>
        <caption>Managing archives with Ark</caption>
        <caption>Training in Artikulate</caption>
        <caption>Using Maxima backend of Cantor</caption>
        <caption>Playing Granatier</caption>
        <caption>Index file manager</caption>
        <caption>2D plot in KAlgebra</caption>
        <caption>Periodic table of elements view in Kalzium</caption>
        <caption>Kdenlive Timeline</caption>
        <caption>Kdenlive Audio Adjustments</caption>
        <caption>Kdenlive Effects</caption>
        <caption>Main view of Kontrast</caption>
        <caption>Main view of Kontrast on mobile</caption>
        <caption>Solitaire selection screen</caption>
        <caption>The main screen in KPhotoAlbum</caption>
        <caption>The typical board position in KSquares</caption>
        <caption>The completed squares in KSquares</caption>
        <caption>An initial board in KSquares</caption>
        <caption>Editing code with KWrite</caption>
        <caption>Basic visualization of data</caption>
        <caption>Basic visualization of data</caption>
        <caption>Advanced layouting of plots and images</caption>
        <caption>Theming</caption>
        <caption>Visualization of mathematical functions</caption>
        <caption>Fit pre-defined or custom models to the data</caption>
        <caption>Smooth the data with the help of various smoothing methods</caption>
        <caption>Notebook interface for interactive sessions in Maxima, Octave, R, Sage, Scilab, Python, Julia, etc</caption>
        <caption>Signal analysis</caption>
        <caption>Herzsprung-Russel diagram in LabPlot</caption>
        <caption>Visualization of data calculated in an interactive Maxima session</caption>
        <caption>Earth globe in Marble</caption>
        <caption>Training with Minuet</caption>
        <caption>Reading manual in Okular</caption>
        <caption>Choosing a collection for training in Parley</caption>
        <caption>The Pix music player</caption>
        <caption>Working with graphs in Rocs</caption>
        <caption>Main screen</caption>
        <caption>The Legend of Zelda: Breath of the Wild running in fullscreen on Ryujinx</caption>
        <caption>Metroid Dread running in fullscreen on Ryujinx</caption>
        <caption>Pokémon Legends: Arceus in fullscreen on Ryujinx</caption>
        <caption>Testing SOAP API</caption>
        <caption>Testing REST API</caption>
        <caption>Settings window</caption>
        <caption>Test as HTML form</caption>
        <caption>The TeXworks source and preview windows</caption>
        <caption>Vinegar gameplay of Phantom Forces, an experience on Roblox</caption>
        <caption>The main Plex window showing Movies and TV Shows</caption>
        <caption>Live TV guide</caption>
        <caption>Playback</caption>
        <caption>Onboarding Dialog</caption>
        <caption>Wallet Overlay</caption>
        <caption>Store Page</caption>
        <caption>Library Page</caption>
        <caption>Main window</caption>
        <caption>Main window</caption>
        <caption>The main user interface</caption>
```
